packages/fuels-types/src/param_types.rs

```diff
@@ -68,38 +70,62 @@ impl ParamType {
         }
     }
 
-    pub fn contains_no_nested_vectors(&self) -> bool {
+    pub fn calculate_num_of_elements(&self, bytes: &[u8]) -> Result<usize> {
+        let memory_size = match self {
+            // The `Bytes` type in the VM uses byte-packing, don't multiply by WORD_SIZE
+            ParamType::Bytes => Ok(ParamType::U8.compute_encoding_width()),
```

Comment:
I don't like the idea that `U8.compute_encoding_width()` returns 1 WORD (as described in the fn doc) meaning 8 B and we're reinterpreting it as 1 B by deciding not to multiply by `WORD_SIZE`. I'd rather we do something like `std::mem::size_of::<u8>()` which does return the number of bytes and does not confuse the meaning of `compute_encoding_width`.

Reply:
Agreed!